src/SFA.DAS.Employer.PR.Web/Controllers/SelectTrainingProviderController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using FluentValidation;
+using FluentValidation.AspNetCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Employer.PR.Web.Authentication;
+using SFA.DAS.Employer.PR.Web.Infrastructure;
+using SFA.DAS.Employer.PR.Web.Infrastructure.Services;
+using SFA.DAS.Employer.PR.Web.Models;
+using SFA.DAS.Employer.PR.Web.Models.Session;
+
+namespace SFA.DAS.Employer.PR.Web.Controllers;
+
+[Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
+[Route("accounts/{employerAccountId}/providers/new/selectProvider", Name = RouteNames.SelectTrainingProvider)]
+public class SelectTrainingProviderController(ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitViewModel> _validator) : Controller
+{
+    [HttpGet]
+    public Task<IActionResult> Index([FromRoute] string employerAccountId, CancellationToken cancellationToken)
+    {
+        var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
+
+        if (sessionModel == null)
+        {
+            return Task.FromResult<IActionResult>(RedirectToAction("Index", "YourTrainingProviders", new { employerAccountId }));
+        }
+
+        var backLink = SetBackLink(employerAccountId, sessionModel.NumberOfLegalEntities);
+
+        SelectTrainingProviderViewModel model = new SelectTrainingProviderViewModel(backLink!);
+        return Task.FromResult<IActionResult>(View(model));
+    }
+
+    [HttpPost]
+    public Task<IActionResult> Index([FromRoute] string employerAccountId, SelectTrainingProviderSubmitViewModel submitModel, CancellationToken cancellationToken)
+    {
+        var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
+        submitModel.SearchTerm = submitModel.Name;
+        var result = _validator.Validate(submitModel);
+
+        var model = GetViewModel(employerAccountId);
```

Comment:
Move this inside the if condition and instead of rebuilding model in happy path, just redirect to Get

Reply:
good idea, done